File: app-mvp-java/src/main/java/com/example/app_mvp_java/net/ApiService.java
package com.example.app_mvp_java.net;

import com.example.model.BaseResponse;
import com.example.model.Login;

import io.reactivex.Observable;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * @author wandervogel
 * @date 2025-09-22
 * @description RxJava 版 ApiService
 */
public interface ApiService {
    @POST("user/login")
    Observable<BaseResponse<Login>> login(
            @Query("username") String username,
            @Query("password") String password
    );

    @POST("user/register")
    Observable<BaseResponse<Login>> register(
            @Query("username") String username,
            @Query("password") String password,
            @Query("repassword") String repassword
    );
}

File: common/src/main/java/com/example/common/net/ApiService.kt
package com.example.common.net

import com.example.common.net.bean.BaseResponse
import com.example.common.net.bean.User
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST

// ApiService.kt
interface ApiService {
    // 退出登录 GET
    @GET("user/logout/json")
    suspend fun logout(): BaseResponse<Any>

    // 注册 POST
    @FormUrlEncoded
    @POST("user/register")
    suspend fun register(
        @Field("username") username: String,
        @Field("password") password: String,
        @Field("repassword") repassword: String
    ): BaseResponse<User>
}